### ISSUE TITLE
no-unnecessary-generics shouldn't error on return types

### ISSUE THREAD

**Issue by cyrilgandon** [palantir/tslint#3252]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.5.2
- __Running TSLint via__: Visual Studio Code

#### TypeScript code being linted

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/69acc07b23b9ef174732a352fc732c363149db15/types/angular/index.d.ts#L1040

```ts
interface IDeferred<T> {
    resolve(value?: T | IPromise<T>): void;
    reject(reason?: any): void;
    notify(state?: any): void;
    promise: IPromise<T>;
}

interface IQService {
    defer<T>(): IDeferred<T>; // error on this line
}
```

with `tslint.json` configuration:

```json
"no-unnecessary-generics": true
```

#### Actual behavior
Error:

>Type parameter is only used once

#### Expected behavior

No error, the return type is an important information.

Given the docs:
>Bad:
>function parse<T>(): T;
>const x = parse<number>();

>Good:
>function parse(): {};
>const x = parse() as number;

Means that instead of doing

    const deferred = $q.defer<string>();

We should doing

    const deferred = $q.defer() as IDeferred<string>;
    // or
    const deferred = <IDeferred<string>>$q.defer();

I find the first syntax fat better than the 2 others. Is this still the official position?

**Comment by cyrilgandon** [comment]
Wrong repo, this rule is from https://github.com/Microsoft/dtslint
